apis/v1alpha1/table.go

```diff
@@ -0,0 +1,180 @@
+	//     in the Amazon Keyspaces Developer Guide.
+	//
+	// The primary key of the table consists of the following columns:
+	//
+	//   - partitionKeys - The partition key can be a single column, or it can
+	//     be a compound value composed of two or more columns. The partition key
+	//     portion of the primary key is required and determines how Amazon Keyspaces
+	//     stores your data.
+	//
+	//   - name - The name of each partition key column.
+	//
+	//   - clusteringKeys - The optional clustering column portion of your primary
+	//     key determines how the data is clustered and sorted within each partition.
+	//
+	//   - name - The name of the clustering column.
+	//
+	//   - orderBy - Sets the ascendant (ASC) or descendant (DESC) order modifier.
+	//     To define a column as static use staticColumns - Static columns store
+	//     values that are shared by all rows in the same partition:
+	//
+	//   - name - The name of the column.
+	//
+	//   - type - An Amazon Keyspaces data type.
+	//
+	// +kubebuilder:validation:Required
+	SchemaDefinition *SchemaDefinition `json:"schemaDefinition"`
+	// The name of the table.
+	// +kubebuilder:validation:Required
+	TableName *string `json:"tableName"`
+	// A list of key-value pair tags to be attached to the resource.
+	//
+	// For more information, see Adding tags and labels to Amazon Keyspaces resources
+	// (https://docs.aws.amazon.com/keyspaces/latest/devguide/tagging-keyspaces.html)
+	// in the Amazon Keyspaces Developer Guide.
+	Tags []*Tag `json:"tags,omitempty"`
+	// Enables Time to Live custom settings for the table. The options are:
+	//
+	//   - status:enabled
+	//
+	//   - status:disabled
+	//
+	// The default is status:disabled. After ttl is enabled, you can't disable it
+	// for the table.
+	//
+	// For more information, see Expiring data by using Amazon Keyspaces Time to
+	// Live (TTL) (https://docs.aws.amazon.com/keyspaces/latest/devguide/TTL.html)
+	// in the Amazon Keyspaces Developer Guide.
+	TTL *TimeToLive `json:"ttl,omitempty"`
+}
+
+// TableStatus defines the observed state of Table
+type TableStatus struct {
+	// All CRs managed by ACK have a common `Status.ACKResourceMetadata` member
+	// that is used to contain resource sync state, account ownership,
+	// constructed ARN for the resource
+	// +kubebuilder:validation:Optional
+	ACKResourceMetadata *ackv1alpha1.ResourceMetadata `json:"ackResourceMetadata"`
+	// All CRS managed by ACK have a common `Status.Conditions` member that
+	// contains a collection of `ackv1alpha1.Condition` objects that describe
+	// the various terminal states of the CR and its backend AWS service API
+	// resource
+	// +kubebuilder:validation:Optional
+	Conditions []*ackv1alpha1.Condition `json:"conditions"`
+	// The unique identifier of the table in the format of an Amazon Resource Name
+	// (ARN).
+	// +kubebuilder:validation:Optional
+	ResourceARN *string `json:"resourceARN,omitempty"`
```

Comment:
This field should be marked as `is_arn` so that the controller stores this information in `ACKResourceMetadata.ARN`

Reply:
I am guessing if I add this logic in the generator.yaml that should do the trick right?
`Table:
    fields:
      ResourceARN:
        is_arn: true`

Reply:
@vishalkatikineni  Exactly :)

Reply:
updated the `generator.yaml` and re-generated using `code-generator`